src/ff1/proptests.rs

```diff
@@ -0,0 +1,78 @@
+use aes::Aes256;
+use num_integer::Integer;
+use proptest::prelude::*;
+
+use super::{BinaryNumeralString, FlexibleNumeralString, FF1};
+
+prop_compose! {
+    fn valid_radix()(radix in 2u32..=(1 << 16)) -> (u32, u16, usize) {
+        let max_numeral = (radix - 1) as u16;
+        let min_len = 0;
+        (radix, max_numeral, min_len)
+    }
+}
+
+prop_compose! {
+    fn flexible_ns()(
+        (radix, max_numeral, min_len) in valid_radix(),
+    )(
+        radix in Just(radix),
```

Comment:
Why is this line necessary? You should be able to just use `radix` directly in the body of the generator.

Reply:
That didn't work for me; the values in the first layer of the strategy aren't accessible beyond the second layer AFAICT.